src/main/Extensions/EmptyTrash/EmptyTrashExtension.ts

```diff
@@ -0,0 +1,116 @@
+
+type Settings = {
+    confirmBeforeEmpty: boolean;
+};
+
+export class EmptyTrashExtension implements Extension {
+    public readonly id = "EmptyTrash";
+    public readonly name = "Empty Trash";
+
+    public readonly nameTranslation = {
+        key: "Empty Trash",
+        namespace: "extension[EmptyTrash]",
+    };
+
+    private readonly defaultSettings: Settings = {
+        confirmBeforeEmpty: true,
+    };
+
+    public readonly author = {
+        name: "Ömer Duran",
+        githubUserName: "omerdduran",
+    };
+
+    public constructor(private readonly assetPathResolver: AssetPathResolver) {}
+
+    public async getSearchResultItems(): Promise<SearchResultItem[]> {
+        return [
+            {
+                name: "Empty Trash",
+                description: "Empty system trash/recycle bin",
+                descriptionTranslation: {
+                    key: "emptyTrashDescription",
+                    namespace: "extension[EmptyTrash]",
+                },
+                id: "emptyTrash:action",
+                image: this.getImage(),
+                defaultAction: {
+                    description: "Empty trash",
+                    descriptionTranslation: {
+                        key: "emptyTrashAction",
+                        namespace: "extension[EmptyTrash]",
+                    },
+                    handlerId: "EmptyTrash",
+                    argument: "empty",
+                    fluentIcon: "DeleteDismissRegular" as FluentIcon,
+                    hideWindowAfterInvocation: true,
+                    requiresConfirmation: true,
+                },
+            },
+        ];
+    }
+
+    public getInstantSearchResultItems(searchTerm: string): SearchResultItem[] {
+        if (!searchTerm.toLowerCase().includes("trash")) {
+            return [];
+        }
+
+        return [
+            {
+                name: "Empty Trash",
+                description: "Empty system trash/recycle bin",
+                descriptionTranslation: {
+                    key: "emptyTrashDescription",
+                    namespace: "extension[EmptyTrash]",
+                },
+                id: "emptyTrash:action",
+                image: this.getImage(),
+                defaultAction: {
+                    description: "Empty trash",
+                    descriptionTranslation: {
+                        key: "emptyTrashAction",
+                        namespace: "extension[EmptyTrash]",
+                    },
+                    handlerId: "EmptyTrash",
+                    argument: "empty",
+                    fluentIcon: "DeleteDismissRegular" as FluentIcon,
+                    hideWindowAfterInvocation: true,
+                    requiresConfirmation: true,
+                },
+            },
+        ];
+    }
+
+    public isSupported(): boolean {
+        return true;
+    }
+
+    public getSettingDefaultValue(key: keyof Settings): boolean {
+        return this.defaultSettings[key];
+    }
```

Comment:
As there is no way to change the settings yet, there is no need to define default settings:

```suggestion
    public getSettingDefaultValue(): undefined {
        return undefined;
    }
```